enarx.ks

```diff
@@ -90,10 +90,88 @@ Type=ether
 [Network]
 DHCP=yes
 EOF
+
+get_maj_min() {
+    local _majmin
+    _majmin="$(stat -L -c '%t:%T' "$1" 2>/dev/null)"
+    printf "%s" "$((0x${_majmin%:*})):$((0x${_majmin#*:}))"
+}
+
+# give the /srv luks partition a partition label
+if ! [[ -b /dev/disk/by-partlabel/srv ]]; then
+   luksdev=/dev/$(ls -1 /sys/dev/block/$(get_maj_min /dev/disk/by-label/srv)/slaves)
+
+   # label the srv partition for later
+   sfdisk --part-label \
+       /dev/block/$(cat /sys/dev/block/$(get_maj_min $luksdev)/../dev) \
+       $(cat /sys/dev/block/$(get_maj_min "$luksdev")/partition) \
+       srv
+else
+   luksdev=/dev/disk/by-partlabel/srv
+fi
+
+# add /srv to /etc/fstab on reinstall
+if ! grep -q "/srv" /etc/fstab; then
+   echo "LABEL=srv /srv xfs defaults 0 0" >> /etc/fstab
+
+   # reconstruct /etc/crypttab
+   for i in /dev/disk/by-uuid/*; do
+       [[ -b "$i" ]] || continue;
+       [[ "$i" -ef "$luksdev" ]] || continue
+       uuid=${i#/dev/disk/by-uuid/}
+       echo "luks-$uuid UUID=$uuid none discard" > /etc/crypttab
+       break
+   done
+
+else
+
+   # umount /srv and close the luks device
+   if umount /srv &>/dev/null || ! mountpoint /srv &>/dev/null; then
+      cryptsetup close /dev/disk/by-label/srv &>/dev/null || :
+   fi
+
+   # bind /srv to the tpm
+   if ! clevis luks unlock -d "$luksdev"; then
+      clevis luks bind -f -k- -d "$luksdev" tpm2 '{}' <<< "temppass"
+      cryptsetup luksRemoveKey "$luksdev" <<< "temppass"
+      clevis luks unlock -d "$luksdev"
+   fi
+   udevadm settle
+   mount /srv
+fi
+
+
+# automatically unlock /srv via clevis
+cat > /etc/systemd/system/local-clevis-luks-askpass.path <<EOF
+[Unit]
+Description=Local Clevis systemd-ask-password Watcher
+DefaultDependencies=no
+Before=local-fs.target
+
+[Path]
+PathChanged=/run/systemd/ask-password
+
+[Install]
+WantedBy=local-fs.target
+EOF
+
+cat > /etc/systemd/system/local-clevis-luks-askpass.service <<EOF
+[Unit]
+Description=Local Clevis LUKS systemd-ask-password Responder
+DefaultDependencies=no
+
+[Service]
+Type=oneshot
+ExecStart=/usr/libexec/clevis-luks-askpass -l
+EOF
+
+systemctl enable local-clevis-luks-askpass.path
```

Comment:
I don't understand why this unit is needed.

Reply:
The original clevis-luks-askpass.path is for `remote-fs.target` only.

Reply:
and we need
```
DefaultDependencies=no
```

Reply:
But the only difference is the `netdev` option in `fstab` and `crypttab`, I'm fairly certain.

Reply:
i.e. Can't you just add `_netdev`?

Reply:
Well, we want to bind to /etc as soon as possible before any daemons are started.. right?
Waiting for remote-fs.target might be too late.